contracts/light-clients/AttestorLightClient.sol

```diff
@@ -0,0 +1,216 @@
+    function getConsensusTimestamp(uint64 revisionHeight) external view returns (uint64) {
+        return _consensusTimestampAtHeight[revisionHeight];
+    }
+
+    /// @inheritdoc ILightClient
+    function updateClient(bytes calldata updateMsg)
+        external
+        notFrozen
+        onlyProofSubmitter
+        returns (ILightClientMsgs.UpdateResult)
+    {
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(updateMsg, (IAttestorMsgs.AttestationProof));
+
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        IAttestorMsgs.StateAttestation memory state =
+            abi.decode(proof.attestationData, (IAttestorMsgs.StateAttestation));
+
+        // Check if height already exists, if it does, check if the timestamp is the same, otherwise revert
+        if (_consensusTimestampAtHeight[state.height] != 0) {
+            if (_consensusTimestampAtHeight[state.height] != state.timestamp) {
+                revert ConflictingTimestamp(state.height, _consensusTimestampAtHeight[state.height], state.timestamp);
+            }
+            return ILightClientMsgs.UpdateResult.NoOp;
+        }
+
+        _consensusTimestampAtHeight[state.height] = state.timestamp;
+        clientState.latestHeight = state.height;
+        return ILightClientMsgs.UpdateResult.Update;
+    }
+
+    /// @inheritdoc ILightClient
+    function verifyMembership(ILightClientMsgs.MsgVerifyMembership calldata msg_)
+        external
+        view
+        notFrozen
+        onlyProofSubmitter
+        returns (uint256)
+    {
+        if (msg_.value.length == 0) revert EmptyValue();
+
+        // Ensure we have a trusted timestamp at the provided height.
+        uint64 height = msg_.proofHeight.revisionHeight;
+        uint64 ts = _consensusTimestampAtHeight[height];
+        if (ts == 0) revert ConsensusTimestampNotFound(height);
+
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(msg_.proof, (IAttestorMsgs.AttestationProof));
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        // Decode the attested packet commitments and verify the attested height matches the provided proof height
+        IAttestorMsgs.PacketAttestation memory packetAttestation =
+            abi.decode(proof.attestationData, (IAttestorMsgs.PacketAttestation));
+
+        if (packetAttestation.height != height) revert HeightMismatch(height, packetAttestation.height);
+
+        // Check membership: value must be present in the attested list
+        bool found = false;
+        bytes32 value = abi.decode(msg_.value, (bytes32));
+        for (uint256 i = 0; i < packetAttestation.packets.length; ++i) {
+            if (packetAttestation.packets[i] == value) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) revert NotMember();
+
+        return uint256(ts);
```

Comment:
This can be made more efficient by returning from within the loop:
```suggestion
        // Check membership: value must be present in the attested list
        bytes32 value = abi.decode(msg_.value, (bytes32));
        for (uint256 i = 0; i < packetAttestation.packets.length; ++i) {
            if (packetAttestation.packets[i] == value) {
                return uint256(ts);
            }
        }

        revert NotMember();
```